include/sys/fs/zfs.h

```diff
@@ -1282,16 +1286,17 @@ typedef enum zfs_ioc {
 	/*
 	 * Linux - 3/64 numbers reserved.
 	 */
-	ZFS_IOC_LINUX = ('Z' << 8) + 0x80,
+	ZFS_IOC_LINUX = ZFS_IOC_FIRST + 0x80,
 	ZFS_IOC_EVENTS_NEXT,			/* 0x5a81 */
 	ZFS_IOC_EVENTS_CLEAR,			/* 0x5a82 */
 	ZFS_IOC_EVENTS_SEEK,			/* 0x5a83 */
-
 	/*
 	 * FreeBSD - 1/64 numbers reserved.
 	 */
-	ZFS_IOC_FREEBSD = ('Z' << 8) + 0xC0,
-
+	ZFS_IOC_FREEBSD = ZFS_IOC_FIRST + 0xc0,
+	ZFS_IOC_NEXTBOOT,		/* 0xc1 */
+	ZFS_IOC_JAIL,			/* 0xc2 */
+	ZFS_IOC_UNJAIL,			/* 0xc3 */
```

Comment:
Before we officially commit this range by using these reserved values let's check with @lundman.  We may want to reserve smaller groups of 32 to leave space for MacOS and windows.

Reply:
Ok, on OSX:
` + 0xD0` and we define one (`ZFS_IOC_PROXY_DATASET`)

On Windows:
` + 0xE0` and we define three (`ZFS_IOC_MOUNT`. `ZFS_IOC_UNMOUNT`, `ZFS_IOC_UNREGISTER_FS`)

Unfortunately, Windows ioctls are a bit weird (they go up by increments of +4), so I had to spell them out:

https://github.com/openzfsonwindows/ZFSin/blob/master/ZFSin/zfs/include/sys/zfs_ioctl.h#L533

The `CTL_CODE` macro shifts the "index" `<< 2`, hence the +4. So the Windows table might have to be separate, unless there is something clever we can do?

Reply:
(Hmm and I should probably fix that lone `ZFS_IOC_RECV_NEW` o_O )

Reply:
Well these don't have to be entirely consistent across platforms, it more for our own convenience when reserving values.  It sound like it may make the most sense for windows to do its own thing when we get to that point.